warehouse/ingest-core/src/main/java/datawave/util/flag/FlagMaker.java

```diff
@@ -507,7 +507,7 @@ protected File write(Collection<InputFile> flagging, FlagDataTypeConfig fc, Stri
             if (fc.getFileListMarker() == null) {
                 String sep = " ";
                 for (InputFile inFile : flagging) {
-                    if (fc.isCollectMetrics())
+                    if (fc.isCollectMetrics() && ctx != null)
```

Comment:
Looking at the history of the FlagMaker.java, I see that the whole concept of metrics got encapsolated into a metrics object of type FlagMetrics.  In there you will see a context and the updating of metrics.  The ctx object in this class should be completely removed, and the calls at like 511 and 531 should instead be calls on the metrics object.